pkg/controller/operators/catalog/operator.go

```diff
@@ -1460,6 +1464,15 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				} else if err != nil {
 					return err
 				} else {
+					// Poll for the CRD to appear
+					logrus.Infof("TEST - About to poll")
+					err = o.pollForCRD(time.Minute, crd.GetName())
```

Comment:
polling here will block the whole control loop - I think we need to exit, requeue the installplan (perhaps with a delay) and then check if the CRD is ready yet.

We do something similar for CSVs here: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/olm/operator.go#L1312-L1317

Reply:
Great point, addressed in the latest version of the PR.